DSCResources/MSFT_NetworkTeam/MSFT_NetworkTeam.psm1

```diff
@@ -0,0 +1,243 @@
+    
+        [Parameter()]
+        [ValidateSet("SwitchIndependent", "LACP", "Static")]
+        [String]$teamingMode = "SwitchIndependent",
+
+        [Parameter()]
+        [ValidateSet("Dynamic", "HyperVPort", "IPAddresses", "MacAddresses", "TransportPorts")]
+        [String]$loadBalancingAlgorithm = "HyperVPort",
+
+        [ValidateSet('Present', 'Absent')]
+        [String]$ensure = 'Present'
+    )
+    Write-Verbose ($localizedData.GetTeamInfo -f $name)
+    $networkTeam = Get-NetLBFOTeam -Name $name -ErrorAction SilentlyContinue
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($networkTeam)
+        {
+            Write-Verbose ($localizedData.foundTeam -f $name)
+            $setArguments = @{
+                'name' = $name
+            }
+
+            if ($networkTeam.loadBalancingAlgorithm -ne $loadBalancingAlgorithm)
+            {
+                Write-Verbose ($localizedData.lbAlgoDifferent -f $loadBalancingAlgorithm)
+                $SetArguments.Add('loadBalancingAlgorithm', $loadBalancingAlgorithm)
+                $isNetModifyRequired = $true
+            }
+
+            if ($networkTeam.TeamingMode -ne $teamingMode)
+            {
+                Write-Verbose ($localizedData.teamingModeDifferent -f $teamingMode)
+                $setArguments.Add('teamingMode', $teamingMode)
+                $isNetModifyRequired = $true
+            }
+            
+            if ($isNetModifyRequired)
+            {
+                Write-Verbose ($localizedData.modifyTeam -f $name)
+                Set-NetLbfoTeam @setArguments -ErrorAction Stop -Confirm:$false
+            }
+
+            $netTeamMembers = Compare-Object -ReferenceObject $teamMembers -DifferenceObject $networkTeam.Members
+            if ($null -ne $netTeamMembers)
+            {
+                Write-Verbose ($localizedData.membersDifferent -f $name)
+                $membersToRemove = ($netTeamMembers | Where-Object {$_.SideIndicator -eq '=>'}).InputObject
+                if ($membersToRemove)
+                {
+                    Write-Verbose ($localizedData.removingMembers -f ($membersToRemove -join ','))
+                    $null = Remove-NetLbfoTeamMember -Name $membersToRemove -Team $name -ErrorAction Stop -Confirm:$false
+                }
+
+                $membersToAdd = ($netTeamMembers | Where-Object {$_.SideIndicator -eq '<='}).InputObject
+                if ($membersToAdd)
+                {
+                    Write-Verbose ($localizedData.addingMembers -f ($membersToAdd -join ','))
+                    $null = Add-NetLbfoTeamMember -Name $membersToAdd -Team $name -ErrorAction Stop -Confirm:$false
+                }
+            }
+            
+        } 
+        else 
+        {
+            Write-Verbose ($localizedData.createTeam -f $name)
+            $null = New-NetLbfoTeam -Name $name -TeamMembers $teamMembers -TeamingMode $teamingMode -LoadBalancingAlgorithm $loadBalancingAlgorithm -ErrorAction Stop -Confirm:$false
```

Comment:
Could you please use some backticks (`) to break this up to multiple lines to keep it under 100 characters?

````powershell
New-NetLbfoTeam `
    -Name $Name `
   -TeamMembers $TeamMembers `
   -TeamingMode $TeamingMode `
   -LoadBalancingAlgorithm $LoadBalancingAlgorithm `
   -ErrorAction Stop `
   -Confirm:$false
````